packages/plugins/experimental/plugin-map/src/components/MapControl.tsx

```diff
@@ -21,29 +22,33 @@ export type Marker = {
 
 const defaults = {
   center: { lat: 37.970833, lng: 23.72611 } as LatLngExpression,
-  zoom: 13,
+  zoom: 1,
 };
 
 /**
  * https://react-leaflet.js.org/docs/api-map
  */
-export const MapControl: FC<{ zoom?: number; center?: LatLngExpression; markers?: Marker[] }> = ({
-  zoom = defaults.zoom,
-  center = defaults.center,
-  markers = [],
-}) => {
+export const MapControl: FC<{ markers?: Marker[] }> = ({ markers = [] }) => {
   const { ref, width, height } = useResizeDetector({ refreshRate: 100 });
   const map = useMap();
+
   useEffect(() => {
     if (width && height) {
       map.invalidateSize();
     }
   }, [width, height]);
+
+  // Set the viewport around the markers, or show the whole world map if `markers` is empty.
   useEffect(() => {
-    if (center) {
-      map.setView(center, zoom ?? map.getZoom() ?? defaults.zoom);
+    if (markers.length > 0) {
+      const bounds = latLngBounds(markers.map((marker) => marker.location));
+      map.fitBounds(bounds);
+    } else {
+      map.setView(defaults.center, defaults.zoom);
     }
-  }, [zoom, center]);
+    // Using plain `[markers]` here causes the effect to trigger extraneously,
+    // overwriting the user's zoom when it shouldn't:
+  }, [markers.length]);
```

Comment:
We could also serialize the markers to JSON here, which would have the advantage of picking up when markers lat/long is changed, not just when they're added/removed.

Reply:
I would do this.

Reply:
If markers is changing too often, that means there's a bug higher up the graph. Why is markers not stable for a constant set of values is the better question?

Reply:
`useMemo` to the rescue